### PR TITLE
msdkenc: add preset target bitrate

### PR DESCRIPTION
**Pull request by mamingy2**
Add preset target bitrate for encoding instead of a fixed one. This result is calculated by linear interpolation of data such as length, width and framerate.

### CODE REVIEW

File: subprojects/gst-plugins-bad/sys/msdk/gstmsdkenc.c [hunk 1]
```diff
@@ -171,13 +171,85 @@ gst_msdkenc_set_context (GstElement * element, GstContext * context)
...
+{
+  mfxInfoMFX *mfx = &thiz->param.mfx;
+  gdouble bitrate = 0;
+  guint low = 0, high = 0;
+  guint pair_cnt = 0;
+  gdouble (*bit_pair)[2] = NULL;
+  gdouble point;
+  guint i;
+
+  static gdouble preset_bitrate[3][5][2] = {
```
Comment (mengker33) [inline subprojects/gst-plugins-bad/sys/msdk/gstmsdkenc.c:185]: Could you add a comment about where these preset values come from (i.e. maybe msdk/onevpl reference)?

File: subprojects/gst-plugins-bad/sys/msdk/gstmsdkenc.c [hunk 2]
```diff
@@ -171,13 +171,85 @@ gst_msdkenc_set_context (GstElement * element, GstContext * context)
...
+        {2058240, 5000 / 1.3}},
+    {{0, 0}, {414720, 12000}}
+  };
+
+  point = mfx->FrameInfo.Width * mfx->FrameInfo.Height *
+      mfx->FrameInfo.FrameRateExtN / mfx->FrameInfo.FrameRateExtD / 30.0;
+  pair_cnt = sizeof (preset_bitrate[0]) / sizeof (gdouble) / 2;
+
+  if (!point) {
+    GST_ERROR_OBJECT (thiz, "Preset liear interpolation failed");
```
Comment (mengker33) [inline subprojects/gst-plugins-bad/sys/msdk/gstmsdkenc.c:197]: Typo? linear...

Comment (mamingy2) [reply subprojects/gst-plugins-bad/sys/msdk/gstmsdkenc.c]: Thanks, done.

File: subprojects/gst-plugins-bad/sys/msdk/gstmsdkenc.c (hunk 1 above)
Comment (mamingy2) [reply subprojects/gst-plugins-bad/sys/msdk/gstmsdkenc.c:188]: Added a comment. Is it acceptable?

**Comment by mamingy2** [comment]
Convert to draft before responce of VPL team.

**Pull request closed without merging**